pkg/metrics/metrics.go

```diff
@@ -56,8 +79,50 @@ func RegisterMetrics() error {
 	return nil
 }
 
-// UpdatePlaceholderGauge ...
-func UpdatePlaceholderGauge() {
+// UpdateSecretsMetrics updates all metrics related to the existance and contents of Secrets
+// used by configure-alertmanager-operator.
+func UpdateSecretsMetrics(list *corev1.SecretList, amconfig alertmanager.Config) {
+
+	// Default to false.
+	pdSecretExists := false
+	dmsSecretExists := false
+	amSecretExists := false
+	metricPDSecretExists.With(prometheus.Labels{"name": config.OperatorName}).Set(float64(0))
```

Comment:
You set this metric potentially twice in this function: here and line 101. I would only set this (and other metrics) one time per run.

Reply:
Yeah, that was intentional with all 5 metrics since I was defaulting them to false. But check out the latest commit and see if you like that better. I changed it to only update each metrics once per run.